ibis/backends/duckdb/__init__.py

```diff
@@ -69,46 +69,213 @@ def _format_kwargs(kwargs: Mapping[str, Any]):
 }
 
 
-class Backend(BaseAlchemyBackend, CanCreateSchema):
+class DuckDBTable(ir.Table):
+    """References a physical table in DuckDB."""
+
+    @property
+    def _client(self):
+        return self.op().source
+
+    @property
+    def name(self):
+        return self.op().name
+
+
+class Backend(BaseBackend, CanCreateSchema):
     name = "duckdb"
-    compiler = DuckDBSQLCompiler
     supports_create_or_replace = True
 
+    def _define_udf_translation_rules(self, expr):
+        # TODO:
+        ...
+
+    def _register_udfs(self, expr):
+        # TODO:
+        ...
+
     @property
     def current_database(self) -> str:
-        return self._scalar_query(sa.select(sa.func.current_database()))
+        return (
+            self.raw_sql("PRAGMA database_size; CALL pragma_database_size();")
+            .arrow()["database_name"]
+            .to_pylist()[0]
+        )
+
+    @property
+    def current_schema(self) -> str:
+        return self.raw_sql("SELECT current_schema()")
+
+    def raw_sql(self, query: str, **kwargs: Any) -> Any:
+        return self.con.execute(query, **kwargs)
```

Comment:
this should probably be `self.con.sql` instead, but we should be able to change it in just the one spot.